contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_encoder_psnr.py

```diff
@@ -0,0 +1,525 @@
+
+    def _extract_metadata_property(self, input: str) -> Dict:
+        properties = {}
+        container_pattern = re.compile(r"container #\d+: (.+)")
+        video_pattern = re.compile(
+            r"video #\d+: (.+)|video\(image\) #\d+: (.+)"
+        )
+        width_pattern = re.compile(r"Width: (\d+)")
+        height_pattern = re.compile(r"Height: (\d+)")
+        frame_rate_pattern = re.compile(r"Frame rate: ([\d/]+)")
+
+        properties["container"] = (
+            container_pattern.search(input).group(1).replace("\n", " ").strip()
+            if container_pattern.search(input)
+            else None
+        )
+        # Check if either video or video(image) pattern matches
+        video_match = video_pattern.search(input)
+        if video_match:
+            properties["video_codec"] = (
+                video_match.group(1).strip()
+                if video_match.group(1)
+                else video_match.group(2).strip()
+            )
+        properties["width"] = (
+            width_pattern.search(input).group(1)
+            if width_pattern.search(input)
+            else None
+        )
+        properties["height"] = (
+            height_pattern.search(input).group(1)
+            if height_pattern.search(input)
+            else None
+        )
+        properties["frame_rate"] = (
+            frame_rate_pattern.search(input).group(1)
+            if frame_rate_pattern.search(input)
+            else None
+        )
+        logging.debug("Prperties got from meta: {}".format(properties))
+        return properties
+
+    def delete_file(self):
+        try:
+            if os.path.exists(self._output_file_full_path):
+                os.remove(self._output_file_full_path)
+        except Exception as e:
+            logging.warn(
+                "Error occurred while deleting file: {}".format(str(e))
+            )
+
+
+class GenioProject(BaseHandler):
+    """
+    Genio project manages platforms and codecs, and handles
+    building.
+
+    Spec: https://download.mediatek.com/aiot/download/release-note/v24.0/v24.0_IoT_Yocto_Feature_Table_v1.0.pdf     # noqa: E501
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initialize the project with a platform and a codec.
+
+        Args:
+            platform (str): The platform name (e.g., 'Platform A').
+            codec (str): The codec name (e.g., 'v4l2h264enc').
+        """
+        super().__init__(**kwargs)
+        self._codec_parser_map = {
+            "v4l2h264enc": "h264parse",
+            "v4l2h265enc": "h265parse",
+        }
+        self._base_pipeline = (
+            "{} filesrc location={} ! decodebin ! videoconvert !"
+            " video/x-raw,format={} ! {}"
+        ).format(
+            GST_LAUNCH_BIN,
+            self._golden_sample,
+            self._color_space,
+            self._codec,
+        )
+
+    def _264_265_command_builder(self) -> List[str]:
```

Comment:
this returns a `str`

Reply:
See https://github.com/canonical/checkbox/pull/1491/commits/18f418371c7f3696951ffcb95a1731d385bd066b